DataTests/CoreDataTestCase.swift

```diff
@@ -8,49 +8,65 @@ import CoreData
 
 class CoreDataTestCase: XCTestCase {
     
-    // Always call super.setUp() in your subclasses to reset the database.
     override func setUp() {
-        // Reset the database.
-        DataController.shared = DataController()
+        super.setUp()
+        
+        // HACK: For some reason switching to in-memory database at launch causes problems with tests,
+        // because some operations are still pending in the app like creating a Domain for localhost
+        // attempting to restore tabs and some shields work.
+        // The workaround is to detect if sqlite database is still used(which means it's first launch of the test suite)
+        // wait a few seconds for it to finish its initialization, then switch the database to in-memory and run
+        // all the tests.
+        //
+        // In the future we should make Data tests more independent of the app state.
+        
+        let storeType = DataController.viewContext.persistentStoreCoordinator!.persistentStores.first!.type
+        if storeType == NSSQLiteStoreType {
```

Comment:
This is a hack but works pretty reliably. I still haven't turned on DataTests for Travis yet.